Various fixes suggested by clippy

mudge opened the pull request:
Running https://github.com/rust-lang-nursery/rust-clippy against the crate revealed a potentially silent lossy cast from u32 to u64.

There's more detail in the commit messages but see the following:

* https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#needless_pass_by_value
* https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#cast_lossless

mudge committed:
Avoid silent lossy cast from u32 to u64

See https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#cast_lossless

mudge commented:
It might also be worth hooking this repository up with [Travis CI](https://travis-ci.org/) and the following `.travis.yml`:

```yaml
language: rust
```

mudge commented on src/lib.rs:
```diff
@@ -38,18 +38,18 @@ impl Lehmer {
         let mut sequence: Vec<u64> = (0..n).collect();
 
         for d in &mut self.code {
             *d = sequence.remove(*d as usize);
```
The risk here is that we are mutating `self.code` in place and then returning an owned copy of it: does that mean it's not possible to call `as_permutation` on the same `Lehmer` twice?

mudge replied:
This makes me think that these really _are_ `to_` methods but it's the mutable reference to `self` that is the problem here. Let me try getting rid of that...

mudge commented on src/lib.rs:
```diff
@@ -38,18 +38,19 @@ impl Lehmer {
-    pub fn to_permutation(mut self) -> Vec<u64> {
-        let n = self.code.len() as u64;
+    pub fn to_permutation(&self) -> Vec<u64> {
+        let mut code = self.code.to_owned();
```
Note this means calling `to_permutation` no longer updates a `Lehmer` instance's `code` in-place: I wasn't sure if this behaviour was intentional or not. If so, we can add tests to ensure that behaviour is preserved (or avoided).

tuzz replied:
Hmm, this behaviour was actually intentional to make it a tiny bit faster: https://github.com/tuzz/lehmer/commit/22166df5ebcc1eac10b5b27244a4eafd303ffd4f

My thinking was the person using this library could `.clone()` the vector first if they wanted to keep it, otherwise it can be happily taken over by this function to gain a little on speed.

The `cargo bench` shows a slight speed drop (~25%) for `#to_permutation`. Maybe instead of changing the signature, I should change the names of these methods? [This page](https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#wrong_self_convention) doesn't seem to have a convention for `mut self` though. 🤔

mudge replied:
Ah, then maybe this is just a naming convention thing: I'll put this change out of the PR.

mudge committed:
Use references instead of passing values

If an argument is passed by value but not consumed in a function, it's
recommended to pass by reference instead.

https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#needless_pass_by_value

tuzz commented:
Thanks @mudge ⭐️
